### PR TITLE
Add missing options to bash completion for `docker build`

### PR DESCRIPTION
**Pull request by albers**
Ref: #15182
This adds the new `--build-arg` and the missing `--disable-content-trust` to bash completion for `docker build`.

Hint: the completion of `--build-arg` is the same as for `docker run -e`.

As the option lists got too long, I refactored it to use the idioms used in completion for `docker run`.

Thanks @sdurrheimer for the ping.
As this PR is not trivial, ping @jfrazelle, @tianon for review.

### CODE REVIEW

**Comment by vdemeester** [comment]
Added this to 1.9.0 milestone (as the command is in there).

**Comment by calavera** [comment]
LGTM

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -461,8 +461,37 @@ _docker_attach() {
 }
 
 _docker_build() {
+	local options_with_args="
+		--build-arg
+		--cgroup-parent
+		--cpuset-cpus
+		--cpuset-mems
+		--cpu-shares -c
```
Comment (tiborvass) [inline contrib/completion/bash/docker]: @albers you should remove `-c`, see #16271

Comment (tiborvass) [inline contrib/completion/bash/docker]: arg this is on build... we need to remove it from build too :(

Comment (vdemeester) [inline contrib/completion/bash/docker]: Handling it here #17119 :wink:

**Comment by tiborvass** [comment]
LGTM

File: contrib/completion/bash/docker (hunk 1 above)
Comment (albers) [inline contrib/completion/bash/docker]: @vdemeester Thanks for taking care of `-c`.

**Pull request merged**